modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,368 @@
+			}
+		}
+		return $content;
+	}
+
+	$filters = array( 'the_content', 'the_excerpt', 'widget_text_content', 'widget_block_content' );
+	foreach ( $filters as $filter ) {
+		add_filter( $filter, 'dominant_color_filter_content_tags', 20 );
+	}
+}
+
+/**
+ * Add CSS needed for to show the dominant color as an image background.
+ *
+ * @since n.e.x.t
+ */
+function dominant_color_add_inline_style() {
+	$handle = 'dominant-color-styles';
+	wp_register_style( $handle, false );
+	wp_enqueue_style( $handle );
+	$custom_css = 'img[data-dominantcolor]:not(.has-transparency) { background-color: var(--dominant-color); background-clip: content-box, padding-box; }';
+	wp_add_inline_style( $handle, $custom_css );
+}
+add_filter( 'wp_enqueue_scripts', 'dominant_color_add_inline_style' );
+
+
+/**
+ * Overloads wp_image_editors() to load the extended classes.
+ *
+ * @since n.e.x.t
+ *
+ * @return string[] Registered image editors class names.
+ */
+function dominant_color_set_image_editors() {
+        if( ! class_exists( 'Dominant_Color_Image_Editor_GD' ) ) {
+	    require_once 'class-dominant-color-image-editor-gd.php';
+	}
+        if( ! class_exists( 'Dominant_Color_Image_Editor_Imagick' ) ) {
+	    require_once 'class-dominant-color-image-editor-imagick.php';
+	}
+
+	return array( 'Dominant_Color_Image_Editor_GD', 'Dominant_Color_Image_Editor_Imagick' );
+}
+
+/**
+ * Get dominant color of image
+ *
+ * @since n.e.x.t
+ *
+ * @param integer $attachment_id the image id.
+ * @return string|null the dominant color of the image. or null if no color is found.
+ */
+function dominant_color_get_dominant_color( $attachment_id ) {
+
+	add_filter( 'wp_image_editors', 'dominant_color_set_image_editors' );
+
+	$file   = get_attached_file( $attachment_id );
+	$editor = wp_get_image_editor( $file );
+
+	if ( ! is_wp_error( $editor ) && method_exists( $editor, 'dominant_color_get_dominant_color' ) ) {
+		$dominant_color = $editor->dominant_color_get_dominant_color();
+		if ( ! is_wp_error( $dominant_color ) ) {
+			return $dominant_color;
+		}
+	}
+
+	remove_filter( 'wp_image_editors', 'dominant_color_set_image_editors' );
+
+	return null;
+}
+
+/**
+ * Works out if color has transparency
+ *
+ * @since n.e.x.t
+ *
+ * @param integer $id the attachment id.
+ * @return bool|null True if the color has transparency, false if it doesn't, null if unknown.
+ */
+function dominant_color_get_has_transparency( $id ) {
+
+	add_filter( 'wp_image_editors', 'dominant_color_set_image_editors' );
+
+	$file = get_attached_file( $id );
+
+	$editor = wp_get_image_editor( $file );
+	remove_filter( 'wp_image_editors', 'dominant_color_set_image_editors' );
+
+	if ( ! is_wp_error( $editor ) && method_exists( $editor, 'dominant_color_get_has_transparency' ) ) {
+		$has_transparency = $editor->dominant_color_get_has_transparency();
+		if ( ! is_wp_error( $has_transparency ) ) {
+
+			return $has_transparency;
+		}
+	}
+
+	return null;
+}
```

Comment:
```suggestion
function dominant_color_get_has_transparency( $id ) {
        $file = get_attached_file( $id );
	add_filter( 'wp_image_editors', 'dominant_color_set_image_editors' );
	$editor = wp_get_image_editor( $file );
	remove_filter( 'wp_image_editors', 'dominant_color_set_image_editors' );

	if ( is_wp_error( $editor ) || !method_exists( $editor, 'dominant_color_get_has_transparency' ) ) {
	      return null;
	}      
	$has_transparency = $editor->dominant_color_get_has_transparency();
	if ( is_wp_error( $has_transparency ) ) {
	      return null;
	}
	
	return $has_transparency;
}
```

Again improve readability.